Lib/ast.py

```diff
@@ -1001,11 +1001,13 @@ def _write_docstring(self, node):
         if node.kind == "u":
             self.write("u")
 
-        # Preserve quotes in the docstring by escaping them
-        value = node.value.replace("\\", "\\\\")
-        value = value.replace('"""', '""\"')
-        if value[-1] == '"':
-            value = value.replace('"', '\\"', -1)
+        value = node.value
+        if value:
+            # Preserve quotes in the docstring by escaping them
+            value = value.replace("\\", "\\\\")
+            value = value.replace('"""', '""\"')
+            if value[-1] == '"':
+                value = value.replace('"', '\\"', -1)
```

Comment:
Do we need to escape also the single quote?

Reply:
I don't think so, we are escaping double quotes because we are printing docstrings with triple doubled quotes, not triple single quotes. I added 2 tests about this.